File: frontend/src/services/allocation.ts
import api from '../utils/api';
import type { MoneyValue, DataSufficiency } from './dashboard';

export interface Advice {
  priority: number;
  title: string;
  amount_suggested: MoneyValue;
  reason: string;
  action_type: 'top_up' | 'pay_extra' | 'hold_buffer' | 'invest' | 'fund_goal' | 'long_term_allocation' | string;
  action_url: string;
}

export interface AllocationAdviceResponse {
  surplus: MoneyValue;
  advices: Advice[];
  data_sufficiency?: DataSufficiency;
  hierarchy?: string[];
  as_of?: string;
  formula_version?: string;
  assumptions?: string[];
}

export const allocationService = {
  async getAllocationAdvice(): Promise<AllocationAdviceResponse> {
    const res = await api.get('/allocation-advice');
    return res.data.data;
  },
};
export default allocationService;

File: backend/internal/dto/allocation.go
package dto

type AdviceDto struct {
	Priority        int        `json:"priority"`
	Title           string     `json:"title"`
	AmountSuggested MoneyValue `json:"amount_suggested"`
	Reason          string     `json:"reason"`
	ActionType      string     `json:"action_type"` // top_up, pay_extra, fund_goal, hold_buffer, invest
	ActionUrl       string     `json:"action_url"`
}

type AllocationAdviceResponse struct {
	Surplus         MoneyValue       `json:"surplus"`
	Advices         []AdviceDto      `json:"advices"`
	DataSufficiency *DataSufficiency `json:"data_sufficiency,omitempty"`
	// Hierarchy documents the decision order used for this response.
	Hierarchy []string `json:"hierarchy"`
}
